core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java

```diff
@@ -185,15 +184,15 @@ public int hashCode() {
     private void checkNotMissingRequired(Result result) {
       if (!result.getValue(Attribute.ID).isPresent()
           || !result.getValue(Attribute.ID).get().getAsString().isPresent()) {
-        throw new RequiredValueMissingException("id is missing in the coordinator state");
+        throw new IllegalArgumentException("id is missing in the coordinator state");
```

Comment:
I'm not sure if `IllegalArgumentException` is appropriate here. It is more like a software's bug, so something different should be used?

Reply:
Thank you for taking a look at this!

Yes, that's true. We should not use `IllegalArgumentException` here. Maybe we can simply use `CoordinatorException` here. I'll fix it. Thanks.